components/server/src/messaging/redis-subscriber.ts

```diff
@@ -0,0 +1,120 @@
+/**
+ * Copyright (c) 2023 Gitpod GmbH. All rights reserved.
+ * Licensed under the GNU Affero General Public License (AGPL).
+ * See License.AGPL.txt in the project root for license information.
+ */
+
+import { RedisClient } from "../redis/client";
+import {
+    HeadlessWorkspaceEventListener,
+    LocalMessageBroker,
+    PrebuildUpdateListener,
+    WorkspaceInstanceUpdateListener,
+} from "./local-message-broker";
+import { inject, injectable } from "inversify";
+import {
+    Disposable,
+    DisposableCollection,
+    RedisWorkspaceInstanceUpdate,
+    WorkspaceInstanceUpdatesChannel,
+} from "@gitpod/gitpod-protocol";
+import { log } from "@gitpod/gitpod-protocol/lib/util/logging";
+import { getExperimentsClientForBackend } from "@gitpod/gitpod-protocol/lib/experiments/configcat-server";
+import { Attributes } from "@gitpod/gitpod-protocol/lib/experiments/types";
+import { reportRedisUpdateCompleted, reportRedisUpdateReceived } from "../prometheus-metrics";
+
+@injectable()
+export class RedisSubscriber implements LocalMessageBroker {
+    constructor(@inject(RedisClient) private readonly redis: RedisClient) {}
+
+    protected workspaceInstanceUpdateListeners: Map<string, WorkspaceInstanceUpdateListener[]> = new Map();
+
+    protected readonly disposables = new DisposableCollection();
+
+    async start(): Promise<void> {
+        const channels = [WorkspaceInstanceUpdatesChannel];
+        const client = this.redis.get();
+
+        for (let chan of channels) {
+            await client.subscribe(chan);
+            this.disposables.push(Disposable.create(() => client.unsubscribe(chan)));
+        }
+
+        client.on("message", async (channel: string, message: string) => {
+            reportRedisUpdateReceived(channel);
```

Comment:
even with feature flag disabled, we still report metrics so we can check the receiver is working